diff --git a/package/boot/uboot-zynq/Makefile b/package/boot/uboot-zynq/Makefile
index c605cfaeb2..9bcacfb62b 100644
--- a/package/boot/uboot-zynq/Makefile
+++ b/package/boot/uboot-zynq/Makefile
@@ -44,11 +44,17 @@ define U-Boot/zybo_z7
   BUILD_DEVICES:=digilent_zynq-zybo-z7
 endef
 
+define U-Boot/ebaz4205
+  NAME:=Ebang EBAZ4205 board
+  BUILD_DEVICES:=ebang_zynq-ebaz4205
+endef
+
 UBOOT_TARGETS := \
 	zc702 \
 	zed \
 	zybo \
-	zybo_z7
+	zybo_z7 \
+	ebaz4205
 
 define Build/InstallDev
 	$(INSTALL_DIR) $(STAGING_DIR_IMAGE)
diff --git a/target/linux/zynq/base-files/etc/board.d/02_network b/target/linux/zynq/base-files/etc/board.d/02_network
index 159b457682..2299c5315a 100644
--- a/target/linux/zynq/base-files/etc/board.d/02_network
+++ b/target/linux/zynq/base-files/etc/board.d/02_network
@@ -8,6 +8,7 @@ case "$(board_name)" in
 avnet,zynq-zed | \
 digilent,zynq-zybo | \
 digilent,zynq-zybo-z7 | \
+ebang,zynq-ebaz4205 | \
 xlnx,zynq-zc702)
 	ucidef_set_interface_lan 'eth0'
 	;;
diff --git a/target/linux/zynq/image/Makefile b/target/linux/zynq/image/Makefile
index 0931871624..e05f830f5c 100644
--- a/target/linux/zynq/image/Makefile
+++ b/target/linux/zynq/image/Makefile
@@ -75,4 +75,11 @@ define Device/xlnx_zynq-zc702
 endef
 TARGET_DEVICES += xlnx_zynq-zc702
 
+define Device/ebang_zynq-ebaz4205
+        $(call Device/FitImageGzip)
+        DEVICE_VENDOR := Ebang
+        DEVICE_MODEL := EBAZ4205
+endef
+TARGET_DEVICES += ebang_zynq-ebaz4205
+
 $(eval $(call BuildImage))
